### PR TITLE
Feature dbi 0 5 db column info

### LINKED ISSUE #96: dbColumnInfo error
### Expected behaviour

From [`DBI.pdf`](https://cran.r-project.org/web/packages/DBI/DBI.pdf):

```r
library(DBI)
conn2 <- dbConnect(RSQLite::SQLite(), ":memory:")
res2 <- dbSendQuery(conn2, "SELECT 1 AS a, 2 AS b")
dbColumnInfo(res2)
#   name    type
# 1    a integer
# 2    b integer
dbFetch(res2)
#   a b
# 1 1 2
dbClearResult(res2)
```

### Actual behaviour

```r
library(DBI)
# ~/sql.yaml defines "SQL_DEV"
conn <- dbConnect(RSQLServer::SQLServer(), server="SQL_DEV", database = "mydatabase")
res <- dbSendQuery(conn, "SELECT 1 AS a, 2 AS b")
dbColumnInfo(res)
# Error in `$<-.data.frame`(`*tmp*`, "field.name", value = "a") : 
#   replacement has 1 row, data has 0
dbFetch(res)
#   a b
# 1 1 2
dbClearResult(res)
# [1] TRUE
```

### Steps to reproduce behaviour

(as above)

```r
dbGetInfo(conn)$db.version
# [1] '12.0.2000'
devtools::session_info()
# Session info -------------------------------------------------------------------
#  setting  value                       
#  version  R version 3.3.1 (2016-06-21)
#  system   x86_64, mingw32             
#  ui       RTerm                       
#  language (EN)                        
#  collate  English_United States.1252  
#  tz       America/Los_Angeles         
#  date     2016-08-22                  
# Packages -----------------------------------------------------------------------
#  package    * version    date       source                                     
#  assertthat   0.1        2013-12-06 CRAN (R 3.3.0)                             
#  DBI        * 0.5        2016-08-22 Github (rstats-db/DBI@1e37697)             
#  devtools     1.12.0     2016-06-24 CRAN (R 3.3.1)                             
#  digest       0.6.10     2016-08-02 CRAN (R 3.3.1)                             
#  dplyr        0.5.0.9000 2016-08-22 Github (javierluraschi/dplyr@6220be8)      
#  evaluate     0.9        2016-04-29 CRAN (R 3.3.0)                             
#  htmltools    0.3.5      2016-03-21 CRAN (R 3.3.0)                             
#  lazyeval     0.2.0      2016-06-12 CRAN (R 3.3.1)                             
#  magrittr     1.5        2014-11-22 CRAN (R 3.3.0)                             
#  memoise      1.0.0      2016-01-29 CRAN (R 3.3.0)                             
#  purrr        0.2.2      2016-06-18 CRAN (R 3.3.1)                             
#  r2         * 0.4.22     2016-08-22 local                                      
#  R6           2.1.3      2016-08-19 CRAN (R 3.3.1)                             
#  Rcpp         0.12.6     2016-07-19 CRAN (R 3.3.1)                             
#  rJava        0.9-8      2016-01-07 CRAN (R 3.3.0)                             
#  rmarkdown    1.0        2016-07-08 CRAN (R 3.3.1)                             
#  RSQLServer   0.2.099    2016-08-22 Github (imanuelcostigan/RSQLServer@1692c2b)
#  stringi      1.1.1      2016-05-27 CRAN (R 3.3.0)                             
#  stringr      1.1.0      2016-08-19 CRAN (R 3.3.1)                             
#  tibble       1.1        2016-07-04 CRAN (R 3.3.1)                             
#  withr        1.0.2      2016-06-20 CRAN (R 3.3.1)                             
#  yaml         2.1.13     2014-06-12 CRAN (R 3.3.0)                             
```

### PR DESCRIPTION
**Pull request by r2evans**
Fixes #96 

Changes proposed:

- add dplyr::bind_rows

Notes: I'm unable to test, since I don't have a local SQL Server instance for the `testthat` scripts to hit against. Therefore, the "check" process does not work locally, so please be patient if my push doesn't work the first time.

@imanuelcostigan

### CODE REVIEW

**Comment by r2evans** [comment]
Sorry the branch name doesn't match your request in 'contributing', I caught that late. I'll do better next time :-)

File: R/dbi-methods.R [hunk 1]
```diff
@@ -530,21 +530,24 @@ setMethod("fetch", c("SQLServerResult", "numeric"),
...
-    data.type = character())
-  if (cols < 1) return(df)
-  for (i in 1:cols) {
-    df$field.name[i] <- rJava::.jcall(res@md, "S", "getColumnName", i)
-    df$field.type[i] <- rJava::.jcall(res@md, "S", "getColumnTypeName", i)
-    ct <- rJava::.jcall(res@md, "I", "getColumnType", i)
-    df$data.type[i] <- jdbcToRType(ct)
+  if (cols < 1L) {
+    df <- data_frame(field.name = character(),
+                     field.type = character(),
```
Comment (imanuelcostigan) [inline R/dbi-methods.R]: I don't align at parentheses after line break. But no need to change this here and below.

**Comment by imanuelcostigan** [comment]
@r2evans My test server is an Windows VM hosted SQL Server kindly provided by Microsoft. I'll check in to see whether I can share details. Note the build is failing. Looks like it Travis CI is having trouble copying encrypted `sql.yaml` file to home directory.

**Comment by imanuelcostigan** [comment]
Looks like the `DBItest::test_meta()` would have picked up this error.

**Comment by r2evans** [comment]
I have no experience with travis builds (other than hoping my PRs don't fail them), so I don't know how to help with the "encrypted `sql.yaml`" problem. Is that failing because of something I did (in the patch), something I haven't done (on my machine), or because the test is being submitted by me and not you?

File: R/dbi-methods.R (hunk 1 above)
Comment (r2evans) [inline R/dbi-methods.R]: Yuck, *how do you work?* ;-)
(jk, I'll adjust future edits ... probably one of the hardest things for me when submitting PRs is accepting the coding standard of the receiving party)
('course, this means I'll have to adjust my emacs indenting rules ... perhaps `local-variables` can be useful here ...)

**Comment by imanuelcostigan** [comment]
No something has changed in the Travis environment which means [one of the commands in the Travis YAML file](https://github.com/imanuelcostigan/RSQLServer/blob/27ab759c02ecbf1ea2f91c4a95674d5a67e825d0/.travis.yml#L5) doesn't successfully execute

**Comment by imanuelcostigan** [comment]
Sorry to do this to you, but are you able to resubmit this patch by branching off master?

**Pull request merged**